cranelift-codegen/src/regalloc/context.rs

```diff
@@ -199,6 +200,15 @@ impl Context {
         self.coloring
             .run(isa, func, domtree, &mut self.liveness, &mut self.tracker);
 
+        // Pass: undo branch splitting where the inserted blocks were not
+        // used by the register allocator.
+        #[cfg(feature = "basic-blocks")]
+        branch_folding::run(func, cfg, domtree, new_blocks);
```

Comment:
I still think this belongs to the branch relaxation part of the code. Doing so as part of the branch relaxation would avoid recomputing the liveness information.